gke/quickstart/autopilot/main.tf

```diff
@@ -0,0 +1,179 @@
+  # accidentally delete this instance by use of Terraform.
+  deletion_protection = false
+}
+
+# Retrieve the access token for configuring the Kubernetes provider
+data "google_client_config" "default" {}
+
+provider "kubernetes" {
+  host                   = "https://${google_container_cluster.default.endpoint}"
+  token                  = data.google_client_config.default.access_token
+  cluster_ca_certificate = base64decode(google_container_cluster.default.master_auth[0].cluster_ca_certificate)
+
+  ignore_annotations = [
+    "^autopilot\\.gke\\.io\\/.*",
+    "^cloud\\.google\\.com\\/.*"
+  ]
+}
+
+resource "kubernetes_deployment_v1" "default" {
+  metadata {
+    name = "example-hello-app-deployment"
+  }
+
+  spec {
+    selector {
+      match_labels = {
+        app = "hello-app"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "hello-app"
+        }
+      }
+
+      spec {
+        container {
+          image = "us-docker.pkg.dev/google-samples/containers/gke/hello-app:2.0"
+          name  = "hello-app-container"
+
+          port {
+            container_port = 8080
+            name           = "hello-app-svc"
+          }
+
+          security_context {
+            allow_privilege_escalation = false
+            privileged                 = false
+            read_only_root_filesystem  = false
+
+            capabilities {
+              add  = []
+              drop = ["NET_RAW"]
+            }
+          }
+
+          liveness_probe {
+            http_get {
+              path = "/"
+              port = "hello-app-svc"
+
+              http_header {
+                name  = "X-Custom-Header"
+                value = "Awesome"
+              }
+            }
+
+            initial_delay_seconds = 3
+            period_seconds        = 3
+          }
+        }
+
+        security_context {
+          run_as_non_root = false
+
+          seccomp_profile {
+            type = "RuntimeDefault"
+          }
+        }
+
+        # Toleration is currently required to prevent perpetual diff:
+        # https://github.com/hashicorp/terraform-provider-kubernetes/pull/2380
+        toleration {
+          effect   = "NoSchedule"
+          key      = "kubernetes.io/arch"
+          operator = "Equal"
+          value    = "amd64"
+        }
+      }
+    }
+  }
+}
+
+# (Optional) Expose the hello-app using a loadblancer
+# resource "kubernetes_service_v1" "default" {
+#   metadata {
+#     name = "example-hello-app-loadbalancer"
+#   }
+
+#   spec {
+#     selector = {
+#       app = kubernetes_deployment_v1.default.spec[0].selector[0].match_labels.app
+#     }
+
+#     ip_family_policy = "RequireDualStack"
+
+#     port {
+#       port        = 80
+#       target_port = kubernetes_deployment_v1.default.spec[0].template[0].spec[0].container[0].port[0].name
+#     }
+
+#     type = "LoadBalancer"
+#   }
+# }
+# [END gke_quickstart_autopilot]
```

Comment:
uncomment ?

Reply:
Thanks @yliaog - Unfortunately GKE/GCP creates Firewall resources OOB, which block the subsequent CI terraform teardown of the network resource.  We'd also probably need to annotate as an internal LB, which isn't as desirable as an example.

Looking at how we can potentially address with documentation.

Reply:
thanks